runtime/src/snapshot_utils.rs

```diff
@@ -861,6 +861,8 @@ pub fn get_bank_snapshots(bank_snapshots_dir: impl AsRef<Path>) -> Vec<BankSnaps
             .for_each(
                 |slot| match BankSnapshotInfo::new_from_dir(&bank_snapshots_dir, slot) {
                     Ok(snapshot_info) => bank_snapshots.push(snapshot_info),
+                    // Ignore any partial snapshots that might be in progress from another thread
+                    Err(SnapshotNewFromDirError::IncompleteDir(_)) => {}
                     Err(err) => debug!("Unable to read bank snapshot for slot {slot}: {err}"),
```

Comment:
Would we want to elevate this back to `warn!()` given that we are now filtering out the known case of one thread seeing an in progress snapshot from another thread ?